### DISCUSSION TITLE
Linear Actuator behaving strangly

### DISCUSSION THREAD

**Discussion by Octal450** [JSBSim-Team/jsbsim#483] [Support requests]
Hi,
I am modelling an analogue HSI instrument by use of a linear actuator, followed by an actuator + lag filter.

I notice some really strange behavior in the linear actuator when there are quick movements about the changeover point specified by the module tag (in my case 359 -> 0 or 0 -> 359

I have observed this affect with the heading bug implementation as well as with the heading card showing current heading, showing the issue is not related to integers vs floating point values.

Properties to take note of:
`heading-input` is magnetic heading in
`heading-linear-cmd` is the linear-actuator

In this image it is working correctly, instead of jumping to 359, it went negative.
![image](https://user-images.githubusercontent.com/25598604/126011391-f5f97da5-d0d2-4320-b2be-46fcf1c032ed.png)

I will now kick the rudder left and right profusely to move the aircraft through the 359 -> 0 or 0 -> 359 transition several times and it works most of the time correctly, then sometimes, it does this:

![image](https://user-images.githubusercontent.com/25598604/126011510-8828735c-b55d-42cf-ab7a-f99d27ee7b01.png)

Huh? it just jumped up to 720 from roughly 0. So now since I have an actuator following the linear actuator, the HSI card rotated (in this case) 2 full revolutions due to the jump from near 0 all the way to 720 (meaning the bug occurred twice, usually its just once) which is the exact behavior I was trying to avoid. This is very strange.

I will also point out it doesn't matter if the linear is currently negative, positive, or with a large number from built up revolutions, or still small near 0 or near +/- 360, the bug still happens.

My code is as follows:
```xml
		<linear_actuator name="/instrumentation/hsi[0]/heading-linear-cmd">
			<input>/instrumentation/hsi[0]/heading-input</input>
			<module>360</module>
		</linear_actuator>
		
		<actuator name="/instrumentation/hsi[0]/heading-linear-actuator">
			<input>/instrumentation/hsi[0]/heading-linear-cmd</input>
			<rate_limit>90</rate_limit>
		</actuator>
```

Am I doing something wrong or did I uncover a bug :D 

Kind Regards,
Josh

**Comment by bcoconni** [comment]
@abassign, any idea about what's going wrong ?

**Comment by bcoconni** [comment]
This is a genuine bug so I have opened the issue #485.

**Reply by Octal450** [reply]
Thanks. Didn't want to open a issue until I was sure it wasn't just a implementation mistake :)

Kind Regards,
Josh

**Reply by bcoconni** [reply]
@Octal450 you did well :+1: 
Thanks for the bug report !
